sdk/azidentity/errors_test.go

```diff
@@ -59,3 +59,48 @@ func TestAuthenticationFailedErrorInterface(t *testing.T) {
 		t.Fatalf("expected troubleshooting link: %s", resBodyString)
 	}
 }
+
+func TestAuthenticationFailedErrorWithoutResponse(t *testing.T) {
+	err := newAuthenticationFailedError(credNameAzureCLI, "error message", nil, nil)
+	if _, ok := err.(*AuthenticationFailedError); !ok {
+		t.Fatalf("expected AuthenticationFailedError, received %T", err)
+	}
+	errMsg := err.Error()
+	if errMsg != "AzureCLICredential: error message" {
+		t.Fatalf("unexpected error message returned: %s", errMsg)
+	}
+}
+
+func TestAuthenticationFailedErrorWithoutOriginalRequest(t *testing.T) {
```

Comment:
Nitpicking: "original" implies we might handle differently some other category of request
```suggestion
func TestAuthenticationFailedErrorWithoutRequest(t *testing.T) {
```